src/libmongoc/src/mongoc/mongoc-collection-private.h

```diff
@@ -39,6 +39,10 @@ struct _mongoc_collection_t {
    bson_t *gle;
 };
 
+struct _mongoc_index_model_t {
+   bson_t *key;
+   mongoc_index_opt_t opts;
```

Comment:
I was thinking it would be a good idea to use this type again.  This will be an opaque type anyway, so it should be safe, and I believe it will allow the flexibility to add new options to that struct without breaking anything.